File: web/tests/reference/intervals/insert-interval.js
function insert(intervals, newInterval) {
  const result = [];
  let [ns, ne] = newInterval;
  const n = intervals.length;
  let i = 0;

  while (i < n && intervals[i][1] < ns) result.push(intervals[i++]);

  while (i < n && intervals[i][0] <= ne) {
    ns = Math.min(ns, intervals[i][0]);
    ne = Math.max(ne, intervals[i][1]);
    i++;
  }
  result.push([ns, ne]);

  while (i < n) result.push(intervals[i++]);

  return result;
}

File: web/tests/reference/intervals/insert-interval.py
def insert(intervals, new_interval):
    result = []
    ns, ne = new_interval
    i, n = 0, len(intervals)

    while i < n and intervals[i][1] < ns:
        result.append(intervals[i])
        i += 1

    while i < n and intervals[i][0] <= ne:
        ns = min(ns, intervals[i][0])
        ne = max(ne, intervals[i][1])
        i += 1
    result.append([ns, ne])

    while i < n:
        result.append(intervals[i])
        i += 1

    return result
